cmd/snap-update-ns/main.go

```diff
@@ -53,12 +56,99 @@ func run() error {
 	if err := parseArgs(os.Args[1:]); err != nil {
 		return err
 	}
+
 	// There is some C code that runs before main() is started.
 	// That code always runs and sets an error condition if it fails.
 	// Here we just check for the error.
 	if err := BootstrapError(); err != nil {
 		return err
 	}
-	// TODO: implement this
-	return fmt.Errorf("not implemented")
+	snapName := opts.Positionals.SnapName
+
+	// Lock the mount namespace so that any concurrently attempted invocations
+	// of snap-confine are synchronized and will see consistent state.
+	lock, err := mount.OpenLock(snapName)
+	if err != nil {
+		return fmt.Errorf("cannot open mount namespace lock file: %s", err)
+	}
+	if err := lock.Lock(); err != nil {
+		return fmt.Errorf("cannot lock mount namespace: %s", err)
+	}
+	defer lock.Close()
+
+	// Read the desired and current mount profiles. Note that missing files
+	// count as empty profiles so that we can gracefully handle a mount
+	// interface connection/disconnection.
+	desiredProfilePath := fmt.Sprintf("%s/snap.%s.fstab", dirs.SnapMountPolicyDir, snapName)
+	desired, err := mount.LoadProfile(desiredProfilePath)
+	if err != nil {
+		return fmt.Errorf("cannot load desired mount profile: %s", err)
+	}
+	fmt.Fprintf(os.Stderr, "DESIRED:\n")
+	for _, entry := range desired.Entries {
+		fmt.Fprintf(os.Stderr, " - %s\n", entry)
+	}
+
+	currentProfilePath := fmt.Sprintf("%s/snap.%s.fstab", dirs.SnapRunNsDir, snapName)
+	current, err := mount.LoadProfile(currentProfilePath)
+	if err != nil {
+		return fmt.Errorf("cannot load current mount profile: %s", err)
+	}
+	fmt.Fprintf(os.Stderr, "CURRENT (before):\n")
+	for _, entry := range current.Entries {
+		fmt.Fprintf(os.Stderr, " - %s\n", entry)
+	}
+
+	// Compute the needed changes and perform each change if needed, collecting
+	// those that we managed to perform or that were performed already.
+	changesNeeded := mount.NeededChanges(current, desired)
+	fmt.Fprintf(os.Stderr, "CHANGES NEEDED:\n")
+	for _, change := range changesNeeded {
+		fmt.Fprintf(os.Stderr, " - %s\n", change)
```

Comment:
How about a small lambda to avoid the repetitions of `fmt.Fprintf(os.Stderr, " - %s\n"....` above and below? The lambda could possibly replace the entire loop, but I'm not sure of that.